riscv/insns/c_sspush_x1.h

```diff
@@ -0,0 +1,32 @@
+require_extension(EXT_ZICFISS);
+require_extension(EXT_ZCA);
+
+reg_t s = STATE.mstatus->read();
+reg_t prev_prv = get_field(s, MSTATUS_MPP);
+reg_t mSSE = get_field(STATE.menvcfg->read(), MENVCFG_SSE);
+
+if (prev_prv != PRV_M && !mSSE)
+  throw trap_illegal_instruction(insn.bits());
+
+if (!p->extension_enabled('S'))
+  throw trap_illegal_instruction(insn.bits());
+else if (prev_prv == PRV_U && !get_field(STATE.senvcfg->read(), SENVCFG_SSE))
+  throw trap_illegal_instruction(insn.bits());
+else if (prev_prv == PRV_S && STATE.v && !get_field(STATE.henvcfg->read(), HENVCFG_SSE))
+  throw trap_illegal_instruction(insn.bits());
+else if (prev_prv == PRV_U && STATE.v && !get_field(STATE.senvcfg->read(), SENVCFG_SSE))
+  throw trap_illegal_instruction(insn.bits());
+
```

Comment:
See comments in c_sspopchk_x5.h

Reply:
I think I can implement the following code.
After checking xSSE, we can do mop/cmop or Zicfiss
```
if (prv == PRV_M)
  xSSE = 0;
else if (prv == PRV_S || prv == PRV_HS)
  xSSE = get_field(STATE.menvcfg->read(), MENVCFG_SSE);
else if (p->extension_enabled('S'))
  xSSE = get_field(STATE.senvcfg->read(), SENVCFG_SSE);
else
  xSSE = false;

// For VS-mode
if (prv == PRV_S && STATE.v)
  xSSE &= get_field(STATE.henvcfg->read(), HENVCFG_SSE);

if (!xSSE) {
  #include "c_mop_N.h"
} else {
  do somthing
}
```

Reply:
For instructions other than `ssrdp` the following suffices
```
if (xSSE) {
  do somthing
}
```
For `ssrdp`, we would include the mop_N.h as you suggested